chore(prismic): add NullContent documentation (#428)
front-commerce/developers.front-commerce.com: source/docs/prismic/resolver-cache.md

  **ProTip:** `contentProperty` can also be used with `singleton` Content.
</blockquote>

### NullContent

> The `NullContent` has been introduced to allow caching of resolvers that return `null`, lets say that you return a tag based on an id, but you only want to return the tags which are in the allow list, in this case, we will return `null` for the tags that are not in the allow list.

Reviewer: split in two sentences : the feature and the example
```suggestion
The `NullContent` has been introduced to allow caching of resolvers that return `null`.

Lets say that you return a tag based on an id, but you only want to return the tags which are in the allow list, in this case, we will return `null` for the tags that are not in the allow list.
```

```js
const allowList = ["tag1", "tag2"];

export default {
  Query: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.TagLoader.loadById(args.id);
      return allowList.includes(tag.name) ? tag : null;
    }),
  },
};
```

> The issue we will have here is that there is no way creating a reference to the Tag document in prismic, so we won't be able to know that `tag3` or `tag4` should cache `null`, so we introduced the NullContent to handle this.

Reviewer: ```suggestion
The issue we have here is that there is no way to create a reference to the Tag document in prismic, so we won't be able to know that `tag3` or `tag4` should cache `null`.

The NullContent handles this.
```

The `NullContent` takes in a single parameter, `contentOrDcoumentId`, which is either the Content object that should be cached as `null`, or the direct id to the document parent document.

> ```js
> import NullContent from "prismic/server/domain/NullContent";
> 
> const allowList = ["tag1", "tag2"];
> 
> export default {
>   Query: {
>     tag: PrismicCachedResolver((parent, args, { loaders }) => {
>       const tag = loaders.TagLoader.loadById(args.id);
>       return allowList.includes(tag.name) ? tag : NullContent(tag);
>     }),
>   },
> };
> ```

Reviewer: I would say the most important here is the difference with the first code sample
```suggestion
```diff
import NullContent from "prismic/server/domain/NullContent";

const allowList = ["tag1", "tag2"];

export default {
  Query: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.TagLoader.loadById(args.id);
-      return allowList.includes(tag.name) ? tag : null;
+      return allowList.includes(tag.name) ? tag : NullContent(tag);
    }),
  },
};
```
```

The direct id of the document is usefull when the caching is based on the parent of the document, for example when you want to cache the `tag` inside a `Faq` document.

```js

> const allowList = ["tag1", "tag2"];
> 
> export default {
>   Fax: {
>     tag: PrismicCachedResolver((parent, args, { loaders }) => {
>       const tag = loaders.DocumentLoader.loadTagByDocumentID(
>         source.documentMetadata.documentID
>       );
>       return allowList.includes(tag.name)
>         ? tag
>         : NullContent(source.documentMetadata.documentID);
>     }),
>   },
> };
> ```

Reviewer: same here (either remove the allowList or add also the import) I would go for the smaller solution
```suggestion
...
export default {
  Fax: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.DocumentLoader.loadTagByDocumentID(
        source.documentMetadata.documentID
      );
      return allowList.includes(tag.name)
        ? tag
        : NullContent(source.documentMetadata.documentID);
    }),
  },
};
```
```

```js
const allowList = ["tag1", "tag2"];

export default {
  Query: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.TagLoader.loadById(args.id);
      return allowList.includes(tag.name) ? tag : null;
    }),
  },
};
```

The issue we have here is that there is no way to create a reference to the Tag document in prismic, so we won't be able to know that `tag3` or `tag4` should cache `null`.

The NullContent handles this.

The `NullContent` takes in a single parameter, `contentOrDcoumentId`, which is either the Content object that should be cached as `null`, or the direct id to the document parent document.

```diff
import NullContent from "prismic/server/domain/NullContent";

const allowList = ["tag1", "tag2"];

export default {
  Query: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.TagLoader.loadById(args.id);
-      return allowList.includes(tag.name) ? tag : null;
+      return allowList.includes(tag.name) ? tag : NullContent(tag);
    }),
  },
};

> 

Reviewer: oh the diff didn't keep the closing backquotes

The direct id of the document is usefull when the caching is based on the parent of the document, for example when you want to cache the `tag` inside a `Faq` document.

```js
...
export default {
  Fax: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.DocumentLoader.loadTagByDocumentID(
        source.documentMetadata.documentID
      );
      return allowList.includes(tag.name)
        ? tag
        : NullContent(source.documentMetadata.documentID);
    }),
  },
};

> 

Reviewer: same here, the diff didn't keep the closing backquotes
